sirepo/auth/ldap.py

```diff
@@ -0,0 +1,56 @@
+"""LDAP login
+:copyright: Copyright (c) 2018-2022 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern import pkconfig
+from pykern.pkdebug import pkdlog, pkdp
+from pykern.pkcollections import PKDict
+import ldap
+import sirepo.quest
+
+AUTH_METHOD = "ldap"
+
+AUTH_METHOD_VISIBLE = True
+
+_cfg = None
+
+class API(sirepo.quest.API):
+    @sirepo.quest.Spec("require_cookie_sentinel")
+    def api_authLdapAuthorized(self):
+
+        # TODO(BG): pass simulation type as method parameter to assign to 'type', currently passing simulation_type to req and leaving 'type' False
+        req = self.parse_post(type=False)
+        u = req.req_data.username
+        p = req.req_data.password
+
+        c = ldap.initialize(_cfg.ldap_server)
+
+        if(len(u) > 256 or len(p) > 256):
+            raise Exception("login user/pass greater than 256 chars")
```

Comment:
Never raise Exception. It's useless to the end user. Rather make this an AssertionError. It's probably a program error.

Reply:
Changing to AssertionError.

I read that assertions stop executing the program when raised and exceptions do not. Is this correct and why exceptions are not usefull/fail fast?

Reply:
AssertionError is a subclass of Exception. All exceptions will stop a program if not caught. This will be caught by uri_router and turned into a ServerError reply.